### DISCUSSION TITLE
DHCPv4 only for specific interface with privsep

### DISCUSSION THREAD

**Discussion by guyyur** [rsmarples/dhcpcd#166] [Q&A] [answered]
Hi,

Is configuration of IPv4 only on specific interface supported?
Example:
dhcpcd.conf
```
...
allowinterfaces wan0 lan0 tun0
noipv4
...
interface wan0
  ipv4
  ...
```

With privsep I get "Bad file descriptor" after renew fails.
dhcpcd[1072]: wan0: failed to renew DHCP, rebinding
dhcpcd[1072]: ps_root_recvmsg: Bad file descriptor

With privsep, ctx->udp_wfd is only initialized if global options have DHCPCD_IPV4 so ctx->ufd_wfd is -1 for ps_inet_sendmsg (called from dhcp_sendudp) with above config.
I think ctx->udp_rfd will also be -1 but I haven't verified it.

When not using privsep, dhcp_start1 checks the interface options for DHCPCD_IPV4 so ctx->udp_wfd and ctx->udp_rfd will be initialized as long as there is one interface with ipv4 option set.

**Comment by rsmarples** [comment, marked as answer]
That's a valid point.
I guess that for privsep enabled code we should just have the socket open incase we activate a new interface with ipv4 because by this point we no longer have permission to open the socket.

The user can always have an IPv6 only system by compling out IPv4 if they really want.
